projects/ngneat/transloco/src/lib/transloco.service.ts

```diff
@@ -142,6 +141,20 @@ export class TranslocoService implements OnDestroy {
         loadTranslation = from(loader);
       }
 
+      // When starting to load a first choice language,
+      // reset the failed loading counter.
+      if (isNil(options.failedCounter)) {
+        options.failedCounter = 0;
+      }
+
+      // When starting to load a first choice language,
+      // build the fallback languages once.
+      // The `failedCounter` and `fallbackLangs` are reused
+      // in following `load()` calls via `handleFailure()`.
+      if (options.failedCounter === 0 && !options.fallbackLangs) {
```

Comment:
I want to clarify before fixing this finding.

The `options` here is of type `LoadOptions` which is:

```ts
export type LoadOptions = {
  fallbackLangs?: string[] | null;
  /** @internal */
  failedCounter?: number;
  inlineLoader?: InlineLoader | null;
};
```

Then `options.fallbackLangs` should not be a `string`. Did I misunderstand something here?

And the `load` method will be called recursively in `handleFailure` callback, so I'm guarding the `options.fallbackLangs` is only set on the first call to `load()`.

Reply:
Re the `fallbackLangs` you understood correctly I confused this option with the one passed to the config 😄 

What I meant by unifying the condition:
```ts
if (isNil(options.failedCounter)) {
  options.failedCounter = 0;

  if (!options.fallbackLangs) {
    options.fallbackLangs = this.fallbackStrategy.getNextLangs(path);
  }
}
```

Reply:
@shaharkazaz I suggest to not combine these `if`s. Because occasionally if `options.failedCounter` is initialized to `0` but `options.fallbackLangs` is not initialized, the combined statements will give wrong result. What do you think?

Reply:
In what cases does it initialize to `0`?

Reply:
Currently, it won't be initialized to `0`. So, I can merge the `if`s without error. But I was thinking it should allow a potential `load` user to initialize this value to `0` or `undefined`.
And let me confirm what's your decision here? @shaharkazaz

Reply:
`undefined` is covered by the `isNil` method.

It doesn't make sense for a user to pass this property since this is a piece of internal information passed between load function calls, so let's merge the `if`s 🙂

Reply:
@shaharkazaz Fixed. Thanks for the discussion!